client/src/components/block/IconBlock.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+interface IconBlocknProps {
+  type: string;
+  index?: number; // ol일 때 순서를 위한 prop
+}
+
+export const IconBlock: React.FC<IconBlocknProps> = ({ type, index }) => {
+  let content;
+
+  if (type === "ul") {
+    content = (
+      <div style={{ marginRight: "8px" }}>
+        <span>•</span>
+      </div>
+    );
+  } else if (type === "ol") {
+    content = (
+      <div style={{ marginRight: "8px" }}>
+        <span>{index}.</span>
+      </div>
+    );
+  } else if (type === "checkbox") {
+    content = (
+      <div style={{ marginRight: "8px" }}>
+        <input type="checkbox" />
+      </div>
+    );
+  } else {
+    content = null;
+  }
+
+  return content;
+};
```

Comment:
React.FC 타입이 안정성에 있어서 지양하는게 좋다고 하더라구요..! (자세한건 조사를 못했습니다..)
그리고 type의 경우에도 nocta의 ElementType 을 재사용하면 좋을 것 같은데, 어떻게 생각하시나요?? 

```suggestion
import { ElementType } from "@noctaCrdt/Interfaces";

interface IconBlockProps {
  type: ElementType;
  index?: number;
}

export const IconBlock = ({ type, index = 1 }: IconBlockProps) => {
  const getIcon = () => {
    switch (type) {
      case "ul":
        return "•";
      case "ol":
        return `${index}.`;
      case "checkbox":
        return <input type="checkbox" />;
      default:
        return null;
    }
  };

  const icon = getIcon();
  if (!icon) return null;

  return (
    <div style={{ marginRight: "8px" }}>
      <span>{icon}</span>
    </div>
  );
};

```

 그리고 해당 코드처럼 중복되는 코드를 조금 더 줄일 수 있을 것 같아요!